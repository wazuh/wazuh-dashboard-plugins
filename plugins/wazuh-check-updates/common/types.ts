export enum API_UPDATES_STATUS {
  UP_TO_DATE = 'upToDate',
  AVAILABLE_UPDATES = 'availableUpdates',
  DISABLED = 'disabled',
  ERROR = 'error',
}

export interface ResponseApiAvailableUpdates {
  current_version?: string;
  update_check?: boolean;
  last_available_major?: Update;
  last_available_minor?: Update;
  last_available_patch?: Update;
  last_check_date?: string;
}

export interface ApiAvailableUpdates extends ResponseApiAvailableUpdates {
  api_id: string;
  status: API_UPDATES_STATUS;
  error?: {
    title?: string;
    detail?: string;
  };
}

export interface Update {
  description: string;
  published_date: string;
  semver: {
    major: number;
    minor: number;
    patch: number;
  };
  tag: string;
  title: string;
}

export interface UserPreferencesDimissedUpdate {
  api_id: string;
  last_major?: string;
  last_minor?: string;
  last_patch?: string;
}

export interface UserPreferences {
  last_dismissed_updates?: UserPreferencesDimissedUpdate[];
  hide_update_notifications?: boolean;
}

export interface AvailableUpdates {
  apis_available_updates: ApiAvailableUpdates[];
  last_check_date: Date;
}

export type savedObjectType = AvailableUpdates | UserPreferences;
