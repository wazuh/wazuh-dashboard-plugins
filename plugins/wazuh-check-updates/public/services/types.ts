import { CtiStatus } from '../shared-components/cti-subscription/types';

export interface apiInfo {
  affected_items: Array<{
    subscription: {
      status: CtiStatus;
    };
  }>;
}
