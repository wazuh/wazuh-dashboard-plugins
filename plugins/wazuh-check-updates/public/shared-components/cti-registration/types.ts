export interface LinkCtiProps {
  handleModalToggle: () => void;
  handleStatusModalToggle?: () => void;
}

export enum CtiStatus {
  AVAILABLE = 'available',
  PENDING = 'pending',
  DENIED = 'denied',
  POLLING = 'polling',
}

export interface StatusCtiRegistrationProps {
  isActive: CtiStatus;
  checkCtiStatus: () => Promise<void>;
}

export interface StatusCtiModalProps {
  handleStatusModalToggle: () => void;
  checkCtiStatus: () => Promise<void>;
  isActive: CtiStatus;
}
