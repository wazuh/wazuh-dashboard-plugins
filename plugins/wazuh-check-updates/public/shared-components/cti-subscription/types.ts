import { Dispatch, SetStateAction } from 'react';

export interface LinkCtiProps {
  handleModalToggle: () => void;
  handleStatusModalToggle?: () => void;
}

export enum CtiStatus {
  ACTIVE = 'active',
  PENDING = 'pending',
  INACTIVE = 'inactive',
}

export interface StatusCtiSubscriptionProps {
  isActive: CtiStatus;
  checkCtiStatus: () => Promise<void>;
}

export interface StatusCtiModalProps {
  handleStatusModalToggle: () => void;
  checkCtiStatus: () => Promise<void>;
  isActive: CtiStatus;
}
