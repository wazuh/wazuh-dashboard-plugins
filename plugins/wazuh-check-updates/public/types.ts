import { WazuhCorePluginStart } from '../../wazuh-core/public';
import { AvailableUpdates } from '../common/types';

export interface WazuhCheckUpdatesPluginSetup {}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface WazuhCheckUpdatesPluginStart {
  UpdatesNotification: () => JSX.Element | null;
  getAvailableUpdates: (
    queryApi: boolean,
    forceQuery: boolean,
  ) => Promise<AvailableUpdates>;
  DismissNotificationCheck: () => JSX.Element | null;
}

export interface AppPluginStartDependencies {
  wazuhCore: WazuhCorePluginStart;
}
