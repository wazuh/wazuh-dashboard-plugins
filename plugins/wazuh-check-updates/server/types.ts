import { ISecurityFactory } from '../../wazuh-core/server/services/security-factory';
import { WazuhCorePluginStart } from '../../wazuh-core/server';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface AppPluginStartDependencies {}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface WazuhCheckUpdatesPluginSetup {}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface WazuhCheckUpdatesPluginStart {}

export type PluginSetup = {
  securityDashboards?: {}; // TODO: Add OpenSearch Dashboards Security interface
  wazuhCore: { dashboardSecurity: ISecurityFactory };
};

export interface AppPluginStartDependencies {
  wazuhCore: WazuhCorePluginStart;
}
