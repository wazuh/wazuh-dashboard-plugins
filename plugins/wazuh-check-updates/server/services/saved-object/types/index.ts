export { availableUpdatesObject } from './available-updates';
export { settingsObject } from './settings';
export { userPreferencesObject } from './user-preferences';
