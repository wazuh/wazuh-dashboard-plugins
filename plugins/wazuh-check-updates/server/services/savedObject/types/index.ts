export { availableUpdatesObject } from './availableUpdates';
export { settingsObject } from './settings';
export { userPreferencesObject } from './userPreferences';
