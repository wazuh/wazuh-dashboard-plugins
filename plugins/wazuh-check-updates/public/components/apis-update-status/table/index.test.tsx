import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ApisUpdateTable } from '.';
import { API_UPDATES_STATUS } from '../../../../common/types';

jest.mock(
  '../../../../../../node_modules/@elastic/eui/lib/services/accessibility/html_id_generator',
  () => ({
    htmlIdGenerator: () => () => 'htmlId',
  })
);

jest.mock('./columns', () => ({
  getApisUpdateStatusColumns: jest.fn().mockReturnValue([
    {
      field: 'api_id',
      name: 'ID',
    },
    {
      field: 'current_version',
      name: 'Version',
    },
    {
      field: 'status',
      name: 'Update status',
    },
  ]),
}));

describe('ApisUpdateTable component', () => {
  test('should return the ApisUpdateTable component', () => {
    const { container, getByText } = render(
      <ApisUpdateTable
        isLoading={false}
        apisAvailableUpdates={[
          {
            api_id: 'api id',
            current_version: '4.3.1',
            status: 'availableUpdates' as API_UPDATES_STATUS,
            last_available_patch: {
              description:
                '## Manager\r\n\r\n### Fixed\r\n\r\n- Fixed a crash when overwrite rules are triggered...',
              published_date: '2022-05-18T10:12:43Z',
              semver: {
                major: 4,
                minor: 3,
                patch: 8,
              },
              tag: 'v4.3.8',
              title: 'Wazuh v4.3.8',
            },
          },
        ]}
      />
    );

    expect(container).toMatchSnapshot();

    expect(getByText('api id')).toBeInTheDocument();
    expect(getByText('4.3.1')).toBeInTheDocument();
    expect(getByText('availableUpdates')).toBeInTheDocument();
  });
});
