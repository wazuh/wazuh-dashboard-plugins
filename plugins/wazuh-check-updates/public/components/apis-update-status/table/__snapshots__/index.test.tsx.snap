// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApisUpdateTable component should return the ApisUpdateTable component 1`] = `
<div>
  <div
    class="euiBasicTable"
  >
    <div>
      <div
        class="euiTableHeaderMobile"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          />
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          />
        </div>
      </div>
      <table
        class="euiTable euiTable--responsive euiTable--auto"
        id="htmlId"
        tabindex="-1"
      >
        <caption
          class="euiScreenReaderOnly euiTableCaption"
        />
        <thead>
          <tr>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_api_id_0"
              role="columnheader"
              scope="col"
            >
              <span
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="ID"
                >
                  ID
                </span>
              </span>
            </th>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_current_version_1"
              role="columnheader"
              scope="col"
            >
              <span
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="Version"
                >
                  Version
                </span>
              </span>
            </th>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_status_2"
              role="columnheader"
              scope="col"
            >
              <span
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="Update status"
                >
                  Update status
                </span>
              </span>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="euiTableRow"
          >
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                ID
              </div>
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                >
                  api id
                </span>
              </div>
            </td>
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Version
              </div>
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                >
                  4.3.1
                </span>
              </div>
            </td>
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Update status
              </div>
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                >
                  availableUpdates
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
